### PR TITLE
BPF: report non-ready when failing to program BPF programs

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Found that the health tests were flaking in BPF mode due because:

- BPF mode was reporting ready even though it was failing to program BPF programs.
- That confused the health tests.  They saw felix ready and did things such as removing the iptables binary to cause felix to fail.  Felix didn't fail in the expected way because it wasn't getting past the BPF programming stage.
- The health tests were generally getting confused by Felix being ready at start of day and then adding additional non-ready health reporters. 

Fix these by:

- Adding health reporting to the BPF endpoint manager so that it reports non-ready if it fails to program BPF programs (e.g. due to lack of host IP).
- Adding a Node in the health tests.
- Waiting for the internal dataplane health report to show up before checking health.

## Related issues/PRs

CORE-10174

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Fix that Felix could briefly report "ready" in the middle of initialisation, before going "non-ready" again until the dataplane was in-sync.  In eBPF mode, Felix will now report non-Ready if it fails to program some BPF programs.  Previously, this would only be reported through logging.  
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Review by tomastigera — APPROVED** [review]
lgmt, some trivial nits

File: felix/dataplane/linux/bpf_ep_mgr.go
Comment (fasaxc) [reply felix/dataplane/linux/bpf_ep_mgr.go:500]: The Register method uses  the struct to mean "supports Readiness/Liveness" rather than "this is the current status"  I think it defaults to false anyway but I was wanting to set the detail message.

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Review by tomastigera — DISMISSED** [review]

**Review by tomastigera — APPROVED** [review]

**Pull request merged**
